### PR TITLE
[Feature] added UserController, updated CarController

### PR DESCRIPTION
**Pull request by philsch91**
- GET /cars and cars/{id} includes now a hourly price for the returned cars
- added GET /user to get user settings
- added PUT /user with payload {"defaultCurrency":"USD"} to update user settings

### CODE REVIEW

File: src/main/java/pacApp/pacController/CarController.java [hunk 1]
```diff
@@ -49,14 +52,16 @@ public CarController(CarRepository repository, RentalRepository rentalRepository
             }
         }
 
-        //TODO: duplicate car locations
+        //TODO: fix duplicate car locations
```
Comment (JamHaven) [inline src/main/java/pacApp/pacController/CarController.java:55]: Please create an issue for this

File: src/main/java/pacApp/pacController/CarController.java [hunk 2]
```diff
@@ -96,8 +103,12 @@ public Car updateCar(@RequestBody Car newCar, @PathVariable Long id){
     }
 
     @DeleteMapping("/cars/{id}")
-    public void deleteCar(@PathVariable Long id){
-        this.repository.deleteById(id);
+    public ResponseEntity deleteCar(@PathVariable Long id){
+        //TODO: check for user role
```
Comment (JamHaven) [inline src/main/java/pacApp/pacController/CarController.java:107]: Create an issue for this please

File: src/main/java/pacApp/pacController/RentalController.java [hunk 3]
```diff
@@ -288,9 +288,12 @@ public ResponseEntity updateRental(@RequestBody Booking booking, @PathVariable L
     }
 
     @DeleteMapping("/rental/{id}")
-    public void deleteRental(@PathVariable Long id){
-        //TODO: check for super user rights
+    public ResponseEntity deleteRental(@PathVariable Long id){
+        //TODO: check for user roles
```
Comment (JamHaven) [inline src/main/java/pacApp/pacController/RentalController.java:292]: Also an issue for this 🗡️

File: src/main/java/pacApp/pacController/UserController.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        GenericResponse response = new GenericResponse(HttpStatus.OK.value(),"Invalid user");
+        return new ResponseEntity<>(userInfo, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/user", method = RequestMethod.PUT,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity updateUser(@RequestBody UserInfo userInfo) {
+        if (userInfo == null) {
+            GenericResponse response = new GenericResponse(400, "Incorrect rental id");
```
Comment (JamHaven) [inline src/main/java/pacApp/pacController/UserController.java]: Rental Id? don't you mean user info?

File: src/main/java/pacApp/pacModel/Currency.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package pacApp.pacModel;
+
+public enum Currency {
+    USD, EUR;
```
Comment (JamHaven) [inline src/main/java/pacApp/pacModel/Currency.java:4]: Why does the backend know USD as well as EUR? Why not more currenies, or just USD?

**Review by JamHaven — APPROVED** [review]
Everything fine, just respect the comments please and give me a heads up when the issues are created.

File: src/main/java/pacApp/pacController/UserController.java (hunk 4 above)
Comment (philsch91) [reply src/main/java/pacApp/pacController/UserController.java]: My fault. Fixed in commit [98dba93](https://github.com/JamHaven/se-carrental-webservice/pull/20/commits/98dba93dcd164b86e9b8db3fb636780f53ac6b2c)

File: src/main/java/pacApp/pacController/CarController.java (hunk 2 above)
Comment (philsch91) [reply src/main/java/pacApp/pacController/CarController.java:107]: I have created issue #21.

File: src/main/java/pacApp/pacController/RentalController.java (hunk 3 above)
Comment (philsch91) [reply src/main/java/pacApp/pacController/RentalController.java:292]: Issue #21 covers this.

File: src/main/java/pacApp/pacModel/Currency.java (hunk 5 above)
Comment (philsch91) [reply src/main/java/pacApp/pacModel/Currency.java:4]: I thought the currencies are needed for an automatic conversion of the car and rental costs based on the default currency of the user for the frontend. Nevertheless, additional currencies are needed to be added. Please correct me @cristianCiu.

**Comment by philsch91** [comment]
This merge has been approved by a verbal confirmation from @JamHaven.

**Pull request merged**